tracer/src/Datadog.Trace/IDatadogTracer.cs

```diff
@@ -19,11 +19,11 @@ internal interface IDatadogTracer
 
         TracerSettings Settings { get; }
 
-        Span StartSpan(string operationName);
+        ISpan StartSpan(string operationName);
```

Comment:
(file-level comment)
As long as we're changing all the public API to use interfaces, should make this `public`, change the type of `Tracer.Instance` to this, and add an `internal Tracer.InternalInstance` (if necessary)?

(and rename this to `ITracer` while we're at it?)

Reply:
> What happens if users swap in a custom ITracer implementation

I'm in favour of adding the public `ITracer` interface, as it makes it easier for users to use the tracer with DI for mocking purposes, but personally I'd make it a vestigial interface that _we_ don't actually use anywhere. So everything else stays the same, none of our methods work with `ITracer` etc. 

It would be literally so they could do something like:

```csharp
var builder = WebApplication.CreateBuilder();
builder.Services.AddSingleton<ITracer>(Tracer.Instance);

var app = builder.Build();

app.MapGet("/", (ITracer tracer) => 
{
   using var scope = tracer.StartActive("helloooooo");
   return "Hello World!" 
});
```

Reply:
Good idea folks. I added an `ITracer` in 4a11cdfc220bd9fbce16445b210660a55e9d7650 , but let me know if we should reduce the API even further